near-sdk-macros/src/core_impl/abi/abi_generator.rs

```diff
@@ -87,13 +87,27 @@ impl ImplItemMethodInfo {
             Some(doc) => quote! { Some(#doc.to_string()) },
             None => quote! { None },
         };
-        let is_view = matches!(&self.attr_signature_info.method_type, &MethodType::View);
-        let is_init = matches!(
-            &self.attr_signature_info.method_type,
-            &MethodType::Init | &MethodType::InitIgnoreState
-        );
-        let AttrSigInfo { is_payable, is_private, is_handles_result, .. } =
-            self.attr_signature_info;
+        let mut modifiers = vec![];
+        let kind = match &self.attr_signature_info.method_type {
+            &MethodType::View => quote! { near_sdk::__private::AbiFunctionKind::View },
+            &MethodType::Regular => {
+                quote! { near_sdk::__private::AbiFunctionKind::Call }
```

Comment:
I think it's worth noting for clarity that until https://github.com/near/near-sdk-rs/pull/890 is merged, the ABI will represent `self`-less functions as call functions. Meaning clients will think they have to send signed transactions and not simply making view calls. Then again, I don't have any stats on how often these are used or if at all, so it might be a trivial nit.

Reply:
Yep!